src/node_javascript.cc

```diff
@@ -6,31 +6,54 @@
 
 namespace node {
 
+#define U(...) __VA_ARGS__
+
 using v8::HandleScope;
 using v8::Local;
 using v8::NewStringType;
 using v8::Object;
 using v8::String;
 
+#define V(id, name_array, data_array)                                         \
+  static const uint8_t id##_name[] = {name_array};                            \
+  static const uint8_t id##_data[] = {data_array};                            \
+  static struct : public String::ExternalOneByteStringResource {              \
+    size_t length() const override {                                          \
+      return sizeof(id##_data);                                               \
+    }                                                                         \
+    const char* data() const override {                                       \
+      return reinterpret_cast<const char*>(id##_data);                        \
+    }                                                                         \
+    void Dispose() override {                                                 \
+      /* Default calls `delete this`. */                                      \
+    }                                                                         \
+  } id##_external_data;
+NODE_NATIVES_MAP(V)
```

Comment:
We don't really seem to have a convention for indentation in macros but two spaces appears to be most common:
```
$ perl -ne 'if (/define V\(/ && /\\$/) { <> =~ /^(\s+)/; print "| ", length($1), " spaces\n" }' src/*.{cc,h} | sort | uniq -c | sort -nr
     16 | 2 spaces
      7 | 4 spaces
      1 | 6 spaces
      1 | 30 spaces
```